python/cudf/cudf/core/frame.py

```diff
@@ -2280,19 +2312,42 @@ def drop_duplicates(
         if len(subset_cols) == 0:
             return self.copy(deep=True)
 
-        result = self.__class__._from_data(
-            *libcudf.stream_compaction.drop_duplicates(
-                self,
-                keys=subset,
-                keep=keep,
-                nulls_are_equal=nulls_are_equal,
-                ignore_index=ignore_index,
-            )
+        keys = self._positions_from_column_names(
+            subset, include_index=not ignore_index
+        )
+        result = self._drop_duplicates(
+            keys=keys,
+            keep=keep,
+            nulls_are_equal=nulls_are_equal,
+            ignore_index=ignore_index,
         )
-
         result._copy_type_metadata(self)
         return result
 
+    def _positions_from_column_names(self, column_names, include_index=False):
+        """Map each column name into their positions in the frame.
+
+        Return positions of the provided column names, offset by the number of
+        index columns if index exists and `include_index` is True. The order
+        of indices returned corresponds to the column order in this Frame.
+        """
+        return [
+            i
+            for i, name in enumerate(self._column_names)
+            if name in set(column_names)
+        ]
+
+    def _drop_duplicates(self, keys, keep, nulls_are_equal, ignore_index):
```

Comment:
Same here, I'd remove the parameter and give it a default in the child class.

Reply:
See https://github.com/rapidsai/cudf/pull/9558#issuecomment-958447946